src/tox/config/sets.py

```diff
@@ -194,6 +194,11 @@ def work_dir_builder(conf: Config, env_name: str | None) -> Path:  # noqa: U100
         def work_dir_post_process(value: Path) -> Path:
             return self._conf.work_dir if self._conf.options.work_dir else value
 
+        def temp_dir_builder(conf: Config, env_name: str | None) -> Path:  # noqa: U100
+            temp_dir = cast(Path, self["work_dir"]) / ".tmp"
+            temp_dir.mkdir(parents=True, exist_ok=True)
+            return temp_dir
```

Comment:
This should be part of the setup and not part of the default handler; where we can also ensure the temp_dir exist if it's user-specified and not existing